enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/OnboardingServiceImpl.java

```diff
@@ -405,11 +382,49 @@ public boolean hasProcessExpired(OnboardingProcessEntity onboardingProcess) {
         // Check expiration for onboarding process due to process timeout
         final Duration processExpiration = onboardingConfig.getProcessExpirationTime();
         final Date createdDateExpirationProcess = DateUtil.convertExpirationToCreatedDate(processExpiration);
-        if (onboardingProcess.getTimestampCreated().before(createdDateExpirationProcess)) {
-            return true;
+        return onboardingProcess.getTimestampCreated().before(createdDateExpirationProcess);
+    }
+
+    private String lookupUser(final String processId, final Map<String, Object> identification) throws OnboardingProcessException {
+        try {
+            final LookupUserRequest lookupUserRequest = LookupUserRequest.builder()
+                    .identification(identification)
+                    .processId(processId)
+                    .build();
+            return onboardingProvider.lookupUser(lookupUserRequest).getUserId();
+        } catch (OnboardingProviderException e) {
+            throw new OnboardingProcessException("User lookup failed, error: " + e.getMessage(), e);
         }
+    }
 
-        return false;
+    @SneakyThrows(OnboardingProcessException.class)
+    private OnboardingProcessEntity createNewProcess(final Map<String, Object> identification, final String identificationData) {
+        final OnboardingProcessEntity process = createNewProcessWithTempUserId(identificationData);
+        logger.debug("Created process ID: {}", process.getId());
+        final String userId = lookupUser(process.getId(), identification);
+        process.setUserId(userId);
+        return process;
     }
 
+    private OnboardingProcessEntity createNewProcessWithTempUserId(final String identificationData) {
+        final OnboardingProcessEntity process = new OnboardingProcessEntity();
+        process.setIdentificationData(identificationData);
+        process.setStatus(OnboardingStatus.ACTIVATION_IN_PROGRESS);
+        process.setUserId("user-id-not-looked-up-yet");
```

Comment:
`userId` has _not-null_ constrain. It is filled later on but still in the same transaction.

Reply:
I don't like it... :) At least, we should introduce a new variable for it. Alternatively, we should consider loosening the constraint.

Reply:
I also don't like this. We should loose the constraint.